app/src/main/java/com/kickstarter/libs/utils/ApplicationLifecycleUtil.java

```diff
@@ -119,6 +123,7 @@ public void onLowMemory() {
   public void onTrimMemory(final int i) {
     if(i == ComponentCallbacks2.TRIM_MEMORY_UI_HIDDEN) {
       this.koala.trackAppClose();
+      this.lake.trackAppClose();
```

Comment:
🤔 If the structure on iOS is:

`Callsite` -> `Koala` -> `TrackingClient` -> `Network`

Is it correct to say the structure on Android is:

`Callsite` -> `TrackingClient` -> `BackgroundService` -> `Network`?

What we did on iOS was to reference two `TrackingClientType`s within `Koala` which allowed us to not change any of the callsites. When we whitelist the events we can then do that in `Koala` and test that it did not send certain events to the `DataLake` `TrackingClientType`.

But in that case I think it would mean referring to two `BackgroundService`s?

Reply:
I think it's this on Android
`Callsite` -> `Koala` -> `TrackingClient` -> `BackgroundService` (handles Network)